src/geometry.rs

```diff
@@ -60,24 +62,26 @@ impl<T> Rect<T> {
 
 impl<T> Rect<T>
 where
-    T: Add<Output = T> + Copy + Clone,
+    T: MaybeMath<T, Option<T>> + Copy + Clone,
+    T: MaybeMath<Option<T>, Option<T>> + Copy + Clone,
+    T: MaybeMath<Option<T>, T> + Copy + Clone,
 {
     /// The sum of [`Rect.start`](Rect) and [`Rect.end`](Rect)
     ///
     /// This is typically used when computing total padding.
     ///
     /// **NOTE:** this is *not* the width of the rectangle.
-    pub(crate) fn horizontal_axis_sum(&self) -> T {
-        self.start + self.end
+    pub(crate) fn horizontal_axis_sum(&self) -> Option<T> {
+        self.start.and_then(|start| start.maybe_add(self.end))
```

Comment:
With `MaybeMath` I should be able to just use `self.start.maybe_add(self.end)` here (?) but not sure exactly how to get that to work. 

Traits this involved are new to me, so bear with me while I try to figure this out 🙃